### ISSUE TITLE
SnykService calls to snyk cli always return true

### ISSUE THREAD

**Issue by renedupont** [opendevstack/ods-jenkins-shared-library#258]
**Describe the bug**
In SnykService there are methods in which the snyk cli is called (auth, test and monitor ...). Apparently they always return 0. Might be the ' | tee report ' part at the end of the script?

**To Reproduce**
Run the pipeline with values that should let these methods return something other than true, e.g. giving a bad snyk auth code.

**Expected behavior**
They should return the right value

**Affected version (please complete the following information):**
master

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
...
[Pipeline] sh (Authenticate with Snyk server)
+ tee -a snyk-report.txt
+ snyk auth TEST
Authentication failed. Please check the API token on https://snyk.io

[Pipeline] echo
Scanning for vulnerabilities with organisation=rened, projectName=test, buildFile=build.gradle, failOnVulnerabilities=false.
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh (Run Snyk test)
+ snyk test --org=rened --file=build.gradle --all-sub-projects
+ tee -a snyk-report.txt
`snyk` requires an authenticated account. Please run `snyk auth` and try again.
...
```

This should have logged 'Snyk auth failed' and ended the stage with error due to the following code:
`    if (!snyk.auth(config.snykAuthenticationCode)) {
      script.error 'Snyk auth failed'
    }`
but it didn't.

**Comment by clemensutschig** [comment]
Rene do you fix this?

**Comment by renedupont** [comment]
Yes @clemensutschig . That PR should fix this issue... BUT... while testing this I uncovered another issue I cannot yet explain. Basically config.failOnVulnerabilities is in [this if statement](https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/src/org/ods/component/ScanWithSnykStage.groovy#L81) always true if it was set via the failOnSnykScanVulnerabilities param which can be set in the pipeline context of the Jenkinsfile, e.g.:
```
odsPipeline(
    image: "${dockerRegistry}/cd/jenkins-slave-maven:2.x",
    projectId: projectId,
    componentId: componentId,
    **failOnSnykScanVulnerabilities: false**,
    sonarQubeBranch: '*',
    branchToEnvironmentMapping: [
        'master': 'test',
        '*'     : 'dev'
    ]
)
```
It is true in this if statement in the snyk stage even if it was set to false in the context. Thus, with my PR for this ticket here the snyk stage will always end in an error once any vulnerability was found no matter to what the failOnSnykScanVulnerabilities was set to. 
Currently, so prior to my PR, it never runs into error since the snyk test script always returns true, even if it shouldn't.

With my PR and when using the recently introduced possibility to set this fail flag as param in to the stage in the Jenkinsfile instead of the former failOnSnykScanVulnerabilities into the context, e.g.:
```
  odsComponentStageScanWithSnyk(context, ['severityThreshold'     : 'medium',
                                          **'failOnVulnerabilities' : false,**
                                          'snykAuthenticationCode': snykAuthenticationCode])
```
everything works fine.

Like mentioned above, I did some troubleshooting but without success yet and I am confused why this is happening when using the old way.
